Fix/xcode build

BlueCocoa opened the pull request:
IMHO, using cmake to generate Xcode project would be better. Imagine if someone uses Linux and writes some code for this project, then he has to edit Xcode project file by hand. 
Even if he leaves us to do this job, it's a little bit complex to maintain Xcode project by hand.

Perhaps a vote?
(+1 for using cmake, -1 for maintaining by hand)

@al1enSuu @Sunbelife @Giolno @ENzero @hem1

BlueCocoa committed:
Using cmake to generate Xcode project

Using cmake to generate Xcode project

Signed-off-by: 0xBBC <0xbbc@0xbbc.com>

BlueCocoa committed:
Add CMake.gitignore

Add
[CMake.gitignore](https://github.com/github/gitignore/blob/master/CMake.
gitignore)

Signed-off-by: 0xBBC <0xbbc@0xbbc.com>

review-ninja commented:
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/79" target="_blank"><img src="https://app.review.ninja/63898156/pull/79/badge" alt="ReviewNinja"/>

0xsuu commented:
0

BlueCocoa committed:
Using build/build_cpp.sh to generate Xcode project

Using build/build_cpp.sh to generate Xcode project.
Usage:
```
./build/build_cpp.sh Xcode
```
Xcode project will generate at ../llama_xcode and will be opened if the
platform is macOS

Signed-off-by: 0xBBC <0xbbc@0xbbc.com>

BlueCocoa committed:
Automatically build Xcode project

Automatically build Xcode project instead of just opening the directory
where the project resides

Signed-off-by: 0xBBC <0xbbc@0xbbc.com>

BlueCocoa committed:
Fixed a bug -- not doing code style checking

BlueCocoa committed:
Fixed a bug when arg is not ("Xcode" | "")

0xsuu commented on README.md:
````diff
@@ -16,3 +16,9 @@ Installation
+### Xcode build
+```
+cd llama_cpp/
+cmake -G Xcode
````
Why build here?

0xsuu commented on build/build_cpp.sh:
```diff
@@ -7,15 +7,28 @@ BASEDIR=$(dirname "$0")
+	# Create Xcode project folder
+	rm -rf ../llama_xcode
+	mkdir -p ../llama_xcode
+	cd ../llama_xcode
```
Use pushd and popd so below code won't be affected.

0xsuu replied:
Can also save some lines of codes as Xcode is a side building target, except from that, every building things should be in build_llama_cpp.

0xsuu commented on build/build_cpp.sh:
```diff
@@ -63,12 +76,24 @@ else
+	make
+fi
+
 if [ $? != 0 ]; then
```
Move this "if" into above else case. And write conditions for Xcode generation failed.

0xsuu commented on build/build_cpp.sh:
```diff
@@ -63,12 +76,24 @@ else
 	else
 		echo "Usage: ./build_cpp.sh 	[-e|--example] [example name]|all"
 		echo "			[-b|--benchmark] [benchmark name]|all"
+		case `uname` in 
```
Consider using uname -s

BlueCocoa committed:
Remove Xcode build section in README.md and make a few changes to build_cpp.sh

Remove Xcode build section in README.md and make a few changes to
build_cpp.sh

Signed-off-by: 0xBBC <0xbbc@0xbbc.com>

0xsuu commented:
LGTM please add instructuon in wiki-quick start

BlueCocoa committed:
Update build_cpp.sh to make sure we'll be in project directory after every if-else block

0xsuu commented:
LGTM

BlueCocoa committed:
Update for coverage

0xsuu commented:
LGTM
